flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/KafkaDeserializationSchemaWrapper.java

```diff
@@ -36,10 +42,36 @@ public class KafkaDeserializationSchemaWrapper<T> implements KafkaDeserializatio
 
     private static final long serialVersionUID = 2651665280744549932L;
 
+    private static Method deserializeWithAdditionalPropertiesMethod = null;
+
+    protected static final String IS_KEY = "IS_KEY";
+
+    protected static final String HEADERS = "HEADERS";
+
+    static {
```

Comment:
Why is https://github.com/apache/flink-connector-kafka/pull/99/files?file-filters%5B%5D=.java&show-viewed-files=true#diff-0a1bb10356e97bd2de4e3e8a257cf8c94290dd0aadb133d9288b6d1bc60b5994R73 different in terms of the `static` block?

You use a static block here, but a static method in DynamicKafkaDeserializationSchema.

However, I like the idea of an interface method `initializeMethod` that returns `Method<DeserializationSchema>` rather than `void` - the typing gives safety that `intializeMethod` returns the Method from a class structure that has Deserialization as the parent.

I do appreciate that the typing doesn't guarantee the method returned is one with the correct parameters - but I think catching the runtime exception is as good as trying to validate the method before using it https://www.tutorialspoint.com/javareflect/javareflect_method_getparametertypes.htm